### PR TITLE
[C-2519] Infinite Scroll Inbox

### PR DESCRIPTION
**Pull request by RoyalSix**
## Description

[Pull Request Description Here]

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> `Fix [#1]()`

### CODE REVIEW

File: packages/react-inbox/src/components/Messages/index.tsx [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
-import TabBar from '../TabBar';
+import TabBar from "../TabBar";
+import {
+  Body, Footer, Header, HeaderText, BodyText, Empty,
+} from "./styled";
 import Loading from "./loading";
-
-import useInbox from "../../hooks/use-inbox";
+import CourierLogo from "./courier_logo_text.svg";
+import { useAtBottom } from "~/hooks/user-at-bottom";
```
Comment (rileylnapier) [inline packages/react-inbox/src/components/Messages/index.tsx:13]: user-at-bottom?  or use-at-bottom?

File: packages/react-inbox/src/components/Messages/index.tsx [hunk 2]
```diff
@@ -18,39 +20,47 @@ const Messages: React.FunctionComponent<InboxProps> = ({
...
+        {isLoading && <Loading />}
         {!isLoading && messages?.length === 0 && (
           <Empty>You have no notifications at this time</Empty>
         )}
       </Body>
       {renderFooter ? renderFooter({}) : (
-      <Footer>
-        <div><span style={{marginTop: 2}}>Powered by&nbsp;&nbsp;</span><CourierLogo /></div>
-      </Footer>)}
+        <Footer>
```
Comment (rileylnapier) [inline packages/react-inbox/src/components/Messages/index.tsx:61]: this should prob be the default "renderFooter" function so people can override it

File: packages/react-inbox/src/hooks/use-messages.ts [hunk 3]
```diff
@@ -22,35 +26,54 @@ export const GET_MESSAGES = `
...
   }
+}
 `;
 
 
 const useMessages = () => {
+  const [startCursor, setStartCursor] = useState(null);
+  const [after, setAfter] = useState(null);
   const inbox = useInbox();
+  const setLoading = useCallback(inbox.setLoading, []);
```
Comment (rileylnapier) [inline packages/react-inbox/src/hooks/use-messages.ts:37]: what makes you want to use `useCallback`?  are you measuring re-renders?

File: packages/react-inbox/src/hooks/user-at-bottom.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import {
+  useCallback, useEffect, useState,
+} from "react";
+
+export function useAtBottom(element) {
```
Comment (rileylnapier) [inline packages/react-inbox/src/hooks/user-at-bottom.ts:5]: oh, decided to use something different from the libraries we shared earlier?

File: packages/react-inbox/src/reducer.ts [hunk 5]
```diff
@@ -1,55 +1,56 @@
...
-        messages: action?.payload?.map(message => ({
-          messageId: message.messageId,
-          created: message.created,
-          title: message?.content?.title,
-          body: message?.content?.body,
-          data: message?.content?.data,
-        })),
-      };
-    }
+  case "inbox/ADD_MESSAGES": {
```
Comment (rileylnapier) [inline packages/react-inbox/src/reducer.ts:25]:
```
case "inbox/ADD_MESSAGES": {
    return {
      ...state,
      startCursor: action?.payload?.startCursor,
      messages: [
        ...(state.messages ?? []),
        ....(action?.payload?.messages?.map(({ node: message }) => ({
            messageId: message.messageId,
            created: message.created,
            title: message?.content?.title,
            body: message?.content?.body,
            data: message?.content?.data,
      }))),
    };
  }

Comment (rileylnapier) [reply packages/react-inbox/src/reducer.ts:25]: or

```case "inbox/ADD_MESSAGES": {
    const newMessages = action?.payload?.messages?.map(({ node: message }) => ({
            messageId: message.messageId,
            created: message.created,
            title: message?.content?.title,
            body: message?.content?.body,
            data: message?.content?.data,
      }));

    return {
      ...state,
      startCursor: action?.payload?.startCursor,
      messages: [
        ...(state.messages ?? []),
        ...newMessages,
    };
  }

File: packages/react-inbox/src/components/Messages/index.tsx (hunk 1 above)
Comment (RoyalSix) [reply packages/react-inbox/src/components/Messages/index.tsx:13]: LOL

File: packages/react-inbox/src/components/Messages/index.tsx (hunk 2 above)
Comment (rileylnapier) [reply packages/react-inbox/src/components/Messages/index.tsx:61]: we can do this later

File: packages/react-inbox/src/reducer.ts (hunk 5 above)
Comment (RoyalSix) [reply packages/react-inbox/src/reducer.ts:25]: Good catch. Immutability

File: packages/react-inbox/src/hooks/user-at-bottom.ts (hunk 4 above)
Comment (RoyalSix) [reply packages/react-inbox/src/hooks/user-at-bottom.ts:5]: Yes, little bit more lightweight and flexible

**Review by rileylnapier — APPROVED** [review]

**Pull request merged**